### PR TITLE
[expo-updates] Break up E2E tests for better reliability

### PR DESCRIPTION
**Pull request by douglowder**
# Why

Attempting to run both basic and assets tests in the same on-success EAS hook is having problems, and occasionally leading to infinite loops until the container is forcibly stopped.

# How

Break up the E2E tests into four workflows, that will each run in a shorter time and more reliably.

# Test Plan

E2E tests should pass reliably.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by Kudo** [comment]
awesome! could you have these tests with [github actions matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs), so we don't have to maintain too many yaml files?

File: packages/expo-updates/CHANGELOG.md [hunk 1]
```diff
@@ -4,12 +4,16 @@
 
 ### 🛠 Breaking changes
 
+- [android] Add support for version 1 of the protocol. ([#20275](https://github.com/expo/expo/pull/20275) by [@wschurman](https://github.com/wschurman))
+
```
Comment (wschurman) [inline packages/expo-updates/CHANGELOG.md:8]: revert

**Review by wschurman — APPROVED** [review]

File: packages/expo-updates/CHANGELOG.md (hunk 1 above)
Comment (douglowder) [reply packages/expo-updates/CHANGELOG.md]: done

**Comment by douglowder** [comment]
@Kudo I'll work on converting these to the actions matrix in a separate PR

**Comment by tsapeta** [comment]
I agree with Kudo and think it should be easily doable even in this PR 🤔

**Comment by douglowder** [comment]
Ok @tsapeta you were right it was easy

**Comment by douglowder** [comment]
Per discussion with @wschurman , separated out the assets test workflow and now it will only run if something changes inside the `expo-updates` module. The basic tests are still run on changes to other modules that might affect updates (the list now includes `@expo/cli` and `@expo/config-plugins`, as it should probably have done always). This should help with the concurrency issues with having 4 jobs running on every PR.

**Pull request merged**